src/adagrams.js

```diff
@@ -1,15 +1,109 @@
+import { LETTER_POOL, SCORE_CHART } from './constant';
+import { cloneObject } from './helpers';
+
+//Implement drawLetters with weighted randomness
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const availableLetters = cloneObject(LETTER_POOL);
+  const selectedLetters = [];
+
+  for (let i = 0; i < 10; i++) {
+    const total = Object.values(availableLetters).reduce((summary, count) => summary + count, 0);
+    const randLetter = Math.floor(Math.random() * total) + 1;
+
+    let cumulative = 0;
+    for (const [letter, count] of Object.entries(availableLetters)) {
+      cumulative += count;
+      if (randLetter <= cumulative) {
+        selectedLetters.push(letter);
+        availableLetters[letter] -= 1;
+        if (availableLetters[letter] === 0) {
+          delete availableLetters[letter];
+        }
+        break;
+      }
+    }
+  }
+
+  return selectedLetters;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const letterCount = {};
+  input = input.toUpperCase();
+
+  for (const letter of lettersInHand) {
+    if (letterCount[letter]) {
+      letterCount[letter] += 1;
+    } else {
+      letterCount[letter] = 1;
```

Comment:
To get the following behavior that we had in Python here in Javascript:
```py
freq_map[item] = freq_map.get(item, 0) + 1
```
We can do the following using the [`||` (OR) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR): 
```js
freqMap[item] = (freqMap[item] || 0) + 1;
```